src/documents/ContextAwareness.md

```diff
@@ -0,0 +1,327 @@
+else some other value held at this path which is to be compared by equality. The `ContextAwareness` API includes two helper functions to assist integrators in issuing contexts of this form, and
+removing them when they are no longer required.
+
+These contexts can be issued with a call of the form:
+
+```
+fluid.contextAware.makeChecks(<checkStructure>);
+```
+
+The `checkStructure` argument is a hash of `contextName` strings to a `checkEntry` record which is documented in the following table:
+
+
+<table>
+    <thead>
+        <tr>
+            <th colspan="3">Members of a <code>checkEntry</code> record supplied as an argument to <code>fluid.contextAware.makeChecks</code></th>
+        </tr>
+        <tr>
+            <th>Member</th>
+            <th>Type</th>
+            <th>Description</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td><code>func</code>/<code>funcName</code></td>
+            <td><code>String</code> <a href="IoCReferences.md">IoC reference</a></td>
+            <td>A function or function name to be evaluated to produce the context value</td>
+        </tr>
+        <tr>
+            <td><code>value</code></td>
+            <td><code>String</code>, <code>Number</code> or <code>Boolean</code></td>
+            <td>The value to be supplied at the <code>value</code> path in the options tructure of the constructed context</td>
+        </tr>
+
+    </tbody>
+</table>
+
+You must supply exactly one of `func`, `funcName` or `value`.
+
+### Example of `fluid.contextAware.makeChecks`
+
+```
+fluid.contextAware.makeChecks({
+    "fluid.browser.supportsBinaryXHR": {
+        funcName: "fluid.enhance.supportsBinaryXHR"
+    },
+    "fluid.browser.supportsFormData": {
+        funcName: "fluid.enhance.supportsFormData"
+    }
+});
+```
+
+In the above example, the two global functions `fluid.enhance.supportsBinaryXHR` and `fluid.enhance.supportsFormData` will be executed, and their
+return values added into contexts with the names `fluid.browser.supportsBinaryXHR` and `fluid.browser.supportsFormData`.
+
+The contexts registered by ``fluid.contextAware.makeChecks`` can be erased from the system by the use of the call 
+
+```
+fluid.contextAware.forgetChecks(`contextNames`);
+```
+
+Here, `contextNames` is can hold either a `String` or `Array of String` holding the keys from the structures previously supplied to `fluid.contextAware.makeChecks`
+
+### Example of `fluid.contextAware.forgetChecks`
+
+For example, the checks registered in the above example `fluid.contextAware.makeChecks` call could be erased by a call to
+
+```
+fluid.contextAware.forgetChecks(["fluid.browser.supportsBinaryXHR", "fluid.browser.supportsFormData"]);
+```
+
+
+## Defining and broadcasting a fresh adaptation in one operation with `fluid.contextAware.makeAdaptation`
+
+A very common use case is to define an adaptation (that is, a `distributeOptions` block which targets the `contextAwareness` area of a collection
```

Comment:
This is a kind of thing that can *cause* an adaptation - that is, an entry in a contextAwareness options section which is acted on. Any other way of getting the options there would do as well...